### DISCUSSION TITLE
Why is approval for different address than the one handling the transfer.

### DISCUSSION THREAD

**Discussion by ZackMcClary34** [0xPolygon/matic.js#268] [Q&A]
I am using the sample node project (https://github.com/maticnetwork/pos-plasma-tutorial) to move some ZkGo from etherium Goerli network to polygon Mumbai network. The "Deposit" feature calls approve to add an allowance for address "0xdD6596F2029e6233DEFfaCa316e6A95217d4Dc34" . But then the "depositFor" that is called uses the "transferFrom" and checks the allowance of address "37c3bfc05d5ebf9ebb3ff80ce0bd0133bf221bc8" . Shouldn't those be the same addresses? Here are some sample transactions moving 0 ZkGo. I was only able to get a transaction of "1" to work by manually approving an allowance for that second address. Please provide an explanation or direct me to help. Thanks so much.

https://goerli.etherscan.io/tx/0xca75510d7a41acfd3933aee7ddf90732ca5aa74a5433e8611ce4afe40022cbf3

https://goerli.etherscan.io/tx/0x60b8fee0eac5f8e93d24200945bbe1bd660306a017297f4f9361bbd9df8a39d1

**Comment by ujjwalguptaofficial** [comment]
this is answered here - https://github.com/maticnetwork/matic.js/issues/269
